ISSUER_PROTOCOL.md

```diff
@@ -0,0 +1,334 @@
+
+
+### Redeeming Tokens
+
+To support the redemption of tokens by the client, server paths that support token redemption should parse the "Sec-Trust-Token" header as a base64 binary blob. This decoded binary blob should be interpreted as a **Trust Token Redemption Request** and passed into the _Redeem_ crypto protocol. The result of _Redeem_ (a **Trust Token Redemption Response**) should be base64 encoded and returned via the "Sec-Trust-Token" header in the HTTP response. Additionally, the issuer should keep track of redeemed tokens to prevent token reuse from malicious clients.
+
+
+#### Redemption Metadata
+
+At redemption time, the metadata encoded in the Trust Token will be embedded in the Signed Redemption Record. The public metadata is embedded via the key ID of the corresponding key used to sign the Trust Token. The private metadata bit is embedded as a one-bit boolean that can be set per redeemer logic and which can be decoded by partners downstream. One potential means of encoding the private metadata is to hash a shared secret and the token hash and take the lowest bit XOR between that and the metadata value (H(shared\_secret||token-hash) ^ private metadata value), allowing only partners with the shared secret to get the true value of the private metadata bit. Other potential schemes involve an additional endpoint which can be used to read the private metadata value based on the Signed Redemption Record (SRR) or token hash.
+
+
+## Trust Token Crypto Protocol
+
+The Trust Token crypto protocol is based on the PMBTokens design in https://eprint.iacr.org/2020/072 (appendix H) using P-384. The necessary keys and function mappings are described below. Serialization follows the TLS presentation language (https://tools.ietf.org/html/rfc8446).
+
+
+### Keys
+
+The Trust Token protocol primarily requires two kinds of keypairs:
+
+*   SRRSigningKey/SRRVerificationKey - A Ed25519 public keypair for signing and verifying the integrity of the Signed Redemption Record response.
+*   TrustTokenSecretKey/TrustTokenPublicKey - A Trust Token keypair used to sign and verify Trust Tokens.
+
+#### SRRSigningKey/SRRVerificationKey
+
+These keys should be generated and stored as standard Ed25519 keys, with the public/verification key being included in the Key Commitment result as a 32-byte RFC8032 encoding.
+
+
+#### TrustTokenSecretKey/TrustTokenPublicKey
+
+These are keys used in Trust Token consisting of elliptic curve scalars and points. All scalars and points are sized based on the curve choice (P-384). Up to 3 keys may be configured in parallel in the key commitment.
+
+Each keypair consist of the following:
+
+
+```
+{
```

Comment:
Guessing these are meant to be the TLS presentation language? In that case, there should be a `struct` before the `{` (ditto throughout), and it's probably worth citing https://tools.ietf.org/html/rfc8446#section-3 the first time you use it.

Also not sure TLS allows `SomeType x, y, z`. You might need them on separate lines.

Reply:
Done.